staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -829,6 +861,9 @@ func (c *Cacher) dispatchEvent(event *watchCacheEvent) {
 			watcher.nonblockingAdd(event)
 		}
 	} else {
+		// Cache serializations of the object across all watchers.
+		setCachingObjects(event, c.versioner)
```

Comment:
if possible, move this to processEvent where we create the `watchCacheEvent{}` so we don't have to worry about races or locking access to modify fields inside the event

Reply:
done [though I also realized we need to create a new event]